src/pudl/load/parquet.py

```diff
@@ -71,11 +30,19 @@ def epacems_to_parquet(df, root_path):
         None
 
     """
+    epacems_pyarrow_schema = (
+        pudl.metadata.classes.Package.from_resource_ids(
```

Comment:
I notice that `Package.from_resource_ids(RESOURCE_METADATA)` is being called in many places throughout the code. Although `pydantic` is fast, this could start to add up, and it may be more clear for the standard package to be computed once and reused. We wanted to avoid validating the metadata on load, with e.g. `from pudl.metadata import PACKAGE`, since then a metadata error would break `import pudl`, making debugging potentially annoying. If this is still a concern, a compromise could be to use a function like `pudl.metadata.build_package(resource_ids=tuple(RESOURCE_METADATA))`, which could easily be cached with `functools.lru_cache` (hence the hashable `tuple` default for `resource_ids`).

Reply:
Calling this repeatedly didn't seem great. Caching it seems like a much better solution.

Reply:
I think I've implemented this as you suggested. I'm curious why you're suggesting it going into the top level of the metadata subpackage, rather than somewhere like `pudl.metadata.helpers`?

I made the default that it would use all of the resources defined in `RESOURCE_METADATA` and am only using the caching function for the times when I'd been building a package with all of the available resources. There are some other times when it's looking at a potentially more dynamically generated set of resource IDs. Do you think they should get switched over too? It seems like it might be confusing to have both the class method and this helper function for building packages.

Why wouldn't we want to use `@lru_cache` on the `Package.from_resource_ids()` class method itself instead?

Reply:
> I'm curious why you're suggesting it going into the top level of the metadata subpackage, rather than somewhere like `pudl.metadata.helpers`?

Mostly because that is where the preloaded `PACKAGE` use to live, and I saw that constant as the culmination of the `metadata` module.

I agree that using the same API throughout would be best. So yes, `@lru_cache` on the `Package.from_resource_ids` class method is probably the cleanest solution. Sorry for the detour!

Reply:
I've added the decorator to `Package.from_resource_ids` and am requiring the `resource_ids` to be a `Tuple` and it seems to be working. I added a default value of `tuple(sorted(RESOURCE_METADATA))` to avoid needing to import that all over the place wherever we want access to the whole PUDL metadata structure.